packages/aila/src/core/lesson/AilaLesson.ts

```diff
@@ -47,6 +51,10 @@ export class AilaLesson implements AilaLessonService {
     this._plan = plan;
   }
 
+  public setPlan(plan: LooseLessonPlan) {
+    this._plan = plan;
+  }
```

Comment:
I think that we could do with some better patterns for getters and setters. For just the plan, we need the class definition, setting the initial value in the initialiser, 2 setter methods and a getter

Something like [parameter properties](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties), but I think that would only work for non-destructured properties